### PR TITLE
Pass the initial URL as an argument to Browser::new, not as a opts property

### LINKED ISSUE #15636: Pass the initial URL as an argument to Browser::new, not as a opts property
_This issue is part of an effort to improve Servo's embedding story. See https://github.com/paulrouget/servoshell/projects/2_

Starting Servo from a third party application requires setting the initial URL as an option:

```
let mut opts = opts::default_opts();
opts.url = ServoUrl::parse(url).ok();
opts::set_defaults(opts);
```

I would expect to be able to do that instead:

```
servo::Browser::new(window, url);
```

### PR DESCRIPTION
**Pull request by yysung1123**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15636 

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15699)

### CODE REVIEW

File: ports/servo/main.rs [hunk 1]
```diff
@@ -142,7 +142,7 @@ fn main() {
     // Our wrapper around `Browser` that also implements some
     // callbacks required by the glutin window implementation.
     let mut browser = BrowserWrapper {
-        browser: Browser::new(window.clone()),
+        browser: Browser::new(window.clone(), opts::get().url.clone().unwrap()),
```
Comment (fabricedesre) [inline ports/servo/main.rs]: what if unwrap() panics?

Comment (deror1869107) [inline ports/servo/main.rs]: So should I add some code like ```if let opts::get().url.clone() = url { }```?

Comment (KiChjang) [inline ports/servo/main.rs]: `.unwrap_or(ServoUrl::parse("about:blank").unwrap())` may make sense here. If that unwrap fails, then we're in deeply screwed territory anyway so it doesn't matter.

**Comment by paulrouget** [comment]
I would not make the url an `Option<>`. Also, ideally, `Opts::url` would disappear. See https://github.com/servo/servo/issues/15636#issuecomment-281239922

ports/servo/main.rs should look at the pref (not opts) and call Browser::new with the right URL. The compositor should not accept an empty URL (create_constellation should not take an Option as an argument).

One of the issue we have here is that command line is handled in components/ (libservo) when I believe should be parsed in ports/ (embedder). Ideally the command line would be handled in ports/ (initialising `opts` and running `Browser::new(…, url)`). It might be beyond the scope of this PR though.

Maybe for now, we should do this:

1. down to create_constellation, the url should never be an option. It's up to main.rs to get the url from 1) command line 2) PREFS 3) about:blank
2. to get the url from the command line, main.rs can use opts, and the would be the only code access opts.url
3. file a follow up issue to make it so that command handling is done in ports/, where all options would be fed to opts, except the url, which would be used to initiate Browser.
4. remove the opts.url access from 2)

File: components/servo/lib.rs [hunk 2]
```diff
@@ -196,7 +196,7 @@ impl<Window> Browser<Window> where Window: WindowMethods + 'static {
         // as the navigation context.
         let (constellation_chan, sw_senders) = create_constellation(opts.user_agent.clone(),
                                                                     opts.config_dir.clone(),
-                                                                    opts.url.clone(),
+                                                                    Some(url),
```
Comment (paulrouget) [inline components/servo/lib.rs]: Can we not use an Option<> here? Is there any situation where we want to initialise a constellation without a url?

File: ports/servo/main.rs [hunk 3]
```diff
@@ -142,7 +143,9 @@ fn main() {
     // Our wrapper around `Browser` that also implements some
     // callbacks required by the glutin window implementation.
     let mut browser = BrowserWrapper {
-        browser: Browser::new(window.clone()),
+        browser: Browser::new(window.clone(),
+                              opts::get().url.clone()
```
Comment (paulrouget) [inline ports/servo/main.rs]: opts.url checks for the PREFS shell.homepage. Shouldn't we remove that from the opts::from_cmdline_args and move it here instead?

**Review by paulrouget — CHANGES_REQUESTED** [review]

File: components/servo/lib.rs (hunk 2 above)
Comment (deror1869107) [inline components/servo/lib.rs]: No. We can change ```fn create_constellation``` to use ```url: ServoUrl``` as parameter

**Comment by paulrouget** [comment]
To this I would add that when starting multiple instances of Servo, we might want to share the options, but not the url.

**Comment by paulrouget** [comment]
@deror1869107, let me know if there's anything I can do to help or if there's anything I could clarify.

**Comment by jdm** [comment]
Closing due to inactivity.

**Pull request closed without merging**